app/controllers/concerns/token_auth.rb

```diff
@@ -36,17 +36,29 @@ def validate_access
   def auth_errors(exception)
     case exception.record
     when :api_token
-      if signed_in?
-        flash[:error] = exception.message
-        render "authentication_error", status: 403
-      else
-        return_to_param = make_return_to("Previous", request.fullpath)
-        session[:return_to] = return_to_param
-        redirect_to root_path(return_to: return_to_param), alert: "Please sign in to complete this action."
-      end
+      render_api_token_exception(exception)
     when Proposal
       render "authorization_error", status: 403, locals: { msg: "You are not allowed to see that proposal." }
     else
+      render_unidentified_exception(exception)
+    end
+  end
+
+  def render_api_token_exception(exception)
+    if signed_in?
+      flash[:error] = exception.message
+      render "authentication_error", status: 403
+    else
+      return_to_param = make_return_to("Previous", request.fullpath)
+      session[:return_to] = return_to_param
+      redirect_to root_path(return_to: return_to_param), alert: "Please sign in to complete this action."
+    end
+  end
+
+  def render_unidentified_exception(exception)
```

Comment:
it's un-identified in the context of the case/when block (it's the default fall-through action). But I take your point. English befuddles.